src/modules/QtQuick/Repeater.js

```diff
@@ -71,6 +71,9 @@ QmlWeb.registerQmlType({
       if (this.$items.length > model) {
         // have more than we need
         this.$removeChildren(model, this.$items.length);
+        // Normally this is done in $insertChildren, but that won't be called
+        // in this case
+        this.count = this.$items.length;
```

Comment:
It looks confusing that `$insertChildren` and `$removeChildren` behave differently here.
As it doesn't seem viable to add this to `$removeChildren` (because `$removeChildren`+`$insertChildren` shouldn't update it twice), it should be removed from `$insertChildren`.

Then, the comment above won't be needed.